File: kinotic-js/workspace/packages/management-api/src/api/model/grind/events/TaskProgressEvent.ts
import { JobRunEventType } from '@/api/model/grind/events/JobRunEventType'

/**
 * A running task reported its progress. Emitted between the task's TaskStartedEvent and its
 * terminal event, as often as the task reports.
 */
export class TaskProgressEvent {

    public readonly type: JobRunEventType.TASK_PROGRESS = JobRunEventType.TASK_PROGRESS

    /**
     * The task's position in the run's task tree.
     */
    public taskPath: string = ''

    /**
     * How close the task is to completion, 0 to 100.
     */
    public percentageComplete: number = 0

    /**
     * What the task is currently doing, or null.
     */
    public message: string | null = null

}

File: kinotic-grind/src/main/java/org/kinotic/grind/api/model/events/TaskProgressEvent.java
package org.kinotic.grind.api.model.events;

import org.kinotic.grind.api.model.ProgressReporter;

/**
 * A running task reported its progress through the {@link ProgressReporter}. Emitted between
 * the task's {@link TaskStartedEvent} and its terminal event, as often as the task reports.
 *
 * @param taskPath           the task's position in the run's task tree
 * @param percentageComplete how close the task is to completion, 0 to 100
 * @param message            what the task is currently doing, or null
 */
public record TaskProgressEvent(String taskPath,
                                int percentageComplete,
                                String message) implements JobRunEvent {
}
